src/image.c

```diff
@@ -472,6 +472,98 @@ static void wolfBoot_verify_signature(uint8_t key_slot,
 }
 #endif /* WOLFBOOT_SIGN_XMSS */
 
+#ifdef WOLFBOOT_SIGN_ML_DSA
+#include <wolfssl/wolfcrypt/dilithium.h>
+static void wolfBoot_verify_signature(uint8_t key_slot,
+        struct wolfBoot_image *img, uint8_t *sig)
+{
+    int       ret = 0;
+    MlDsaKey  ml_dsa;
+    uint8_t * pubkey = NULL;
+    int       pub_len = 0;
+    int       sig_len = 0;
+    int       verify_res = 0;
+
+    wolfBoot_printf("info: ML-DSA wolfBoot_verify_signature\n");
+
+    pubkey = keystore_get_buffer(key_slot);
+
+    if (pubkey == NULL) {
+        wolfBoot_printf("error: ML-DSA pubkey not found\n");
+        return;
+    }
+
+    ret = wc_MlDsaKey_Init(&ml_dsa, NULL, INVALID_DEVID);
+
+    if (ret != 0) {
+        wolfBoot_printf("error: wc_MlDsaKey_Init returned %d\n", ret);
+        return;
+    }
+
+    /* Set the ML-DSA security level. */
+    ret = wc_MlDsaKey_SetParams(&ml_dsa, ML_DSA_LEVEL);
+
+    if (ret != 0) {
+        wolfBoot_printf("error: wc_MlDsaKey_SetParams(%d)" \
+                        " returned %d\n", ML_DSA_LEVEL, ret);
+        return;
```

Comment:
Prefer a single return flow to make sure `wc_MlDsaKey_Free(&ml_dsa);` gets called. Refactor function with `if (ret == 0)` logic and only have a single return at bottom.